lib/index.js

```diff
@@ -264,6 +275,25 @@ exports.init = function (sequelize, optionsArg) {
         log('revisionId', currentRevisionId);
       }
 
+      // Check if all required fields have been provided to the opts / CLS
+      if (options.extraDataFields) {
+        // get all required field keys as an array
+        var requiredFields = _.keys(_.pickBy(options.extraDataFields, function (required) {
+          return required;
+        }));
+        if (requiredFields && requiredFields.length) {
+          var extraData = ns && ns.get(options.extraDataContinuationKey) || opt['extraData'];
```

Comment:
Shouldn't it be `options.extraData` instead of coming from the model options?
```suggestion
          var extraData = ns && ns.get(options.extraDataContinuationKey) || options.extraData;
```